internal/engine/build_and_deployer_live_update_test.go

```diff
@@ -0,0 +1,226 @@
+package engine
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/synclet/sidecar"
+)
+
+type testCase struct {
+	env k8s.Env
+
+	baseManifest        model.Manifest
+	liveUpdSyncs        []model.LiveUpdateSyncStep
+	liveUpdRuns         []model.LiveUpdateRunStep
+	liveUpdRestart      bool
+	fullRebuildTriggers []string
+
+	changedFile string // leave empty for no changed files
+
+	// Docker actions
+	expectDockerBuildCount   int
+	expectDockerPushCount    int
+	expectDockerCopyCount    int
+	expectDockerExecCount    int
+	expectDockerRestartCount int
+
+	// Synclet actions
+	expectSyncletUpdateContainerCount int
+	expectSyncletHotReload            bool
+	expectK8sDeploy                   bool
+	expectSyncletDeploy               bool
+}
+
+func runTestCase(t *testing.T, f *bdFixture, tCase testCase) {
+	var bs store.BuildStateSet
+	if tCase.changedFile != "" {
+		changed := f.WriteFile(tCase.changedFile, "blah")
+		bs = resultToStateSet(alreadyBuiltSet, []string{changed}, f.deployInfo())
```

Comment:
fwiw, the `alreadyBuiltSet` global you use here isn't really generalizable, it really only works with the sancho test data

Reply:
yeah -- are you suggesting i change it, or just an fyi?